docs/recipes/direct-audit/direct-audit.js

```diff
@@ -0,0 +1,84 @@
+ */
+'use strict';
+
+const fs = require('fs');
+
+const makeDevtoolsLog =
+  require('lighthouse/lighthouse-core/test/network-records-to-devtools-log.js');
+const ModuleDuplication = require('lighthouse/lighthouse-core/computed/module-duplication.js');
+const DuplicatedJavascript =
+  require('lighthouse/lighthouse-core/audits/byte-efficiency/duplicated-javascript.js');
+const LegacyJavascript = require('lighthouse/lighthouse-core/audits/legacy-javascript.js');
+
+/**
+ * @param {Array<{url: string, content: string, map: any}>} scriptDatas
+ */
+function createGathererData(scriptDatas) {
+  const SourceMaps = scriptDatas.map(data => {
+    return {
+      scriptUrl: data.url,
+      map: data.map,
+    };
+  });
+  const ScriptElements = scriptDatas.map((data, i) => {
+    return {
+      requestId: `1000.${i}`,
+      src: data.url,
+      content: data.content,
+    };
+  });
+  const networkRecords = scriptDatas.map((data, i) => {
+    return {
+      requestId: `1000.${i}`,
+      url: data.url,
+      content: data.content,
+    };
+  });
+  networkRecords.push({url: 'https://www.example.com', resourceType: 'Document'});
+
+  const artifacts = {
+    URL: {finalUrl: 'https://www.example.com'},
+    devtoolsLogs: {defaultPass: makeDevtoolsLog(networkRecords)},
+    SourceMaps,
+    ScriptElements,
+  };
+
+  return {
+    artifacts,
+    networkRecords,
+  };
+}
+
+async function run() {
+  /* eslint-disable max-len */
+  const FOO_JS_1 = fs.readFileSync(require.resolve('lighthouse/lighthouse-core/test/fixtures/source-maps/coursehero-bundle-1.js'), 'utf-8');
+  const FOO_JS_MAP_1 = JSON.parse(fs.readFileSync(require.resolve('lighthouse/lighthouse-core/test/fixtures/source-maps/coursehero-bundle-1.js.map'), 'utf-8'));
+  const FOO_JS_2 = fs.readFileSync(require.resolve('lighthouse/lighthouse-core/test/fixtures/source-maps/coursehero-bundle-2.js'), 'utf-8');
+  const FOO_JS_MAP_2 = JSON.parse(fs.readFileSync(require.resolve('lighthouse/lighthouse-core/test/fixtures/source-maps/coursehero-bundle-2.js.map'), 'utf-8'));
+  /* eslint-enable max-len */
+
+  const context = {computedCache: new Map()};
+  const {artifacts, networkRecords} = createGathererData([
+    {url: 'https://www.example.com/foo1.js', content: FOO_JS_1, map: FOO_JS_MAP_1},
+    {url: 'https://www.example.com/foo2.js', content: FOO_JS_2, map: FOO_JS_MAP_2},
+  ]);
+
+  // Request a computed artifact directly.
+  const duplication = await ModuleDuplication.request(artifacts, context);
+
+  // Run an audit directly.
+  const legacyJavascriptResult = await LegacyJavascript.audit(artifacts, context);
+
+  // Run a byte-efficiency audit directly.
+  const duplicatedJavascriptResult =
+    await DuplicatedJavascript.audit_(artifacts, networkRecords, context);
```

Comment:
don't you still want to call byte-efficiency audits with `.audit()`? Most people probably wouldn't want the intermediate result.

Reply:
> don't you still want to call byte-efficiency audits with `.audit()`? Most people probably wouldn't want the intermediate result.

speaking of which, we do have the slightly awkward thing that audits return an `LH.Audit.Product`, then it's `Runner` that calls `Audit.generateAuditResult()` to turn that into an `LH.Audit.Result` (which is what you might expect going off of LHR types). @paulirish's idea of a lib for this might be able to hide that away, though (it could really just generate the artifacts and call `-A` mode).

Reply:
> don't you still want to call byte-efficiency audits with .audit()? Most people probably wouldn't want the intermediate result.

Don't know yet - maybe not. Use case is still nebulous. I deferred it for now because `ByteEfficienyAudit.audit` requires a trace.